### PR TITLE
Enchancement : Refactored HomeScreen, TrafficViolation, and RegisterNewVehicle

### PR DESCRIPTION
**Pull request by harismalik-1**
# Description

Refactored three main activities that are HomeScreen, TrafficViolations, and RegisterNewVehicle such that they adhere to the Model View Presenter pattern and the principles of Clean Architecture.

Fixes # (issue)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by ji24077 — APPROVED** [review]
Seems like nicely refactored by following Clean Archteture :)

File: app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoPresenter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.course_project_autotrace.BasicCarInfo;
```
Comment (l0minous) [inline app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoPresenter.java:1]: Splits Basic Car Info into the MVP pattern

File: app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoPresenter.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        this.fetchInfoUseCase = fetchInfoUseCase;
+    }
+
+    // Load the car info method
+    public void loadCarInfo(Intent intent){
+        CarEntity.Car car1 = fetchInfoUseCase.fromIntent(intent);
+        carView.displayCarDetails(car1);
+    }
+
+    // method to navigate back to the LoginScreen
```
Comment (l0minous) [inline app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoPresenter.java:23]: Provides comments explaining what each section of the code does

File: app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoView.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.course_project_autotrace.BasicCarInfo;
```
Comment (l0minous) [inline app/src/main/java/com/course_project_autotrace/BasicCarInfo/BasicCarInfoView.java:1]: Although I was being skeptical of the View File as it is quite short,  we still have to follow clean architecture

File: app/src/main/java/com/course_project_autotrace/BasicCarInfo/CarEntity.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.course_project_autotrace.BasicCarInfo;
```
Comment (l0minous) [inline app/src/main/java/com/course_project_autotrace/BasicCarInfo/CarEntity.java:1]: Creates a new entity class, something I forgot to consider

File: app/src/main/java/com/course_project_autotrace/RegisterNewVehicle/RegisterNewVehicle.java [hunk 5]
```diff
@@ -1,87 +1,70 @@
...
 import android.widget.ImageButton;
 import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
 
+import com.course_project_autotrace.CarInformation.CarInfo;
+import com.course_project_autotrace.ForgotPassword.ForgotPassword1;
 import com.course_project_autotrace.Hompage.HomeScreen;
+import com.course_project_autotrace.Login.LoginScreen;
 import com.course_project_autotrace.R;
-import com.google.firebase.auth.FirebaseAuth;
```
Comment (l0minous) [inline app/src/main/java/com/course_project_autotrace/RegisterNewVehicle/RegisterNewVehicle.java:15]: Cleans Up import statements

File: app/src/main/res/layout/activity_traffic_violations.xml [hunk 6]
```diff
@@ -167,20 +167,6 @@
         app:srcCompat="@drawable/backarrow"
         tools:ignore="RedundantDescriptionCheck" />
 
-    <ImageButton
```
Comment (l0minous) [inline app/src/main/res/layout/activity_traffic_violations.xml:170]: Reimplements the screen without the refresh button as discussed as it was an issue we were facing

**Review by l0minous — APPROVED** [review]
Overall does completes the refactoring of the files we discussed, tested the program and its features, everything works as intended. Great job!

**Pull request merged**